src/components/NewWorkspaceModal.js

```diff
@@ -120,14 +120,19 @@ const NewWorkspaceModal = withDisplayName('NewWorkspaceModal', ({
           const featuredList = await Ajax().FirecloudBucket.getFeaturedWorkspaces()
           Ajax().Metrics.captureEvent(Events.workspaceClone, {
             featured: _.some({ namespace: cloneWorkspace.workspace.namespace, name: cloneWorkspace.workspace.name }, featuredList),
-            fromWorkspaceName: cloneWorkspace.workspace.name, fromWorkspaceNamespace: cloneWorkspace.workspace.namespace,
-            toWorkspaceName: workspace.name, toWorkspaceNamespace: workspace.namespace
+            ...extractCrossWorkspaceDetails(cloneWorkspace, {
+              // Clone response does not include cloudPlatform, cross-cloud cloning is not supported.
+              workspace: _.merge(workspace, { cloudPlatform: cloneWorkspace.workspace.cloudPlatform })
```

Comment:
To minimise the performance penalty of gathering metrics, would it make sense to merge the result of these `extract...` calls with the cloud platform rather than with the (larger) workspace object since you know that that's going to be extracted?

Reply:
I can't imagine that there is any appreciable performance penalty. We are talking about extremely small data structures here, and the merging is not being done repeatedly per event.